plugins/statistics.koplugin/main.lua

```diff
@@ -753,12 +754,28 @@ function ReaderStatistics:getIdBookDB()
         SELECT count(id)
         FROM   book
         WHERE  title = ?
-            AND authors = ?
-            AND md5 = ?;
+          AND  authors = ?
+          AND  md5 = ?;
     ]]
     local stmt = conn:prepare(sql_stmt)
-    local result = stmt:reset():bind(self.data.title, self.data.authors, self.doc_md5):step()
+    local title, authors = self.data.title, self.data.authors
+    local result = stmt:reset():bind(title, authors, self.doc_md5):step()
     local nr_id = tonumber(result[1])
+    if nr_id == 0 and self.ui.paging then -- old strings are null-terminated
+        title = title .. "\0"
```

Comment:
May be make that comment more verbose:
```
In the past, title and/or authors strings, got from MuPDF, may have been or not null terminated.
We need to check with all combinations if a book with these null terminated exists, and use it.
```
Btw, are these auto updated to non-nul-terminated elsewhere ? I see you don't explicitely update them.

Reply:
They remain null-terminated.

Reply:
No real opinion whether we should fix them or let them be, if they are just getting stored as-is and we find them again, and what the user sees is clean.
Any thought about whay may happen with the cloud storage statistics sync if between devices, and one has that book opened with strings nul terminated, and the other device has it without them?

Reply:
I'm not good at sync, but at the first glance
https://github.com/koreader/koreader/blob/ca569c1355d14ad023eb1bcad7b995ae027e423d/plugins/statistics.koplugin/main.lua#L3166-L3171
it's better to keep the strings in the local db unchanged.